### DISCUSSION TITLE
Need help with renovate configuration errors

### DISCUSSION THREAD

**Discussion by yevhen-hryhorevskyi** [renovatebot/renovate#10202] [Request Help]
Hi there, 
I struggled with renovate configuration following the docs. I got validation errors for "Invalid" fields described in the documentation.
What I want to do:
I want renovate to make "auto-merge" to 2 different landing branches:
- `renovate/auto-updates-dev` for any NPM `devDependency` that passed our CI/CD
- `renovate/auto-updates` for any minor/patch NPM `dependency` that passed our CI/CD.
The default branch is protected and PRs cannot be merged without team-member approval.

I came up with following configuration:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    "docker:enableMajor"
  ],
  "prConcurrentLimit": 3,
  "rebaseWhen": "auto",
  "npm": {
    "postUpdateOptions": ["npmDedupe"],
    "packageRules": [
      {
        "description": "automerge devDependencies updates if CI/CD passes",
        "matchDepTypes": ["devDependencies"],
        "matchPackagePatterns": ["*"],
        "automerge": true,
        "automergeType": "branch",
        "baseBranches": ["renovate/auto-updates-dev"]
      },
      {
        "description": "automerge minor & patch dependencies updates if CI/CD passes",
        "matchUpdateTypes": ["minor", "patch"],
        "matchDepTypes": ["dependencies"],
        "matchPackagePatterns": ["*"],
        "automerge": true,
        "automergeType": "branch",
        "baseBranches": ["renovate/auto-updates"]
      }
    ]
  }
}
```
But I'm getting error like:
`Invalid configuration option: npm.packageRules[0].matchDepTypes, Invalid configuration option: npm.packageRules[0].matchPackagePatterns, Invalid configuration option: npm.packageRules[1].matchDepTypes, Invalid configuration option: npm.packageRules[1].matchPackagePatterns, Invalid configuration option: npm.packageRules[1].matchUpdateTypes, npm.packageRules: Each packageRule must contain at least one selector (paths, languages, baseBranchList, managers, datasources, depTypeList, packageNames, packagePatterns, excludePackageNames, excludePackagePatterns, sourceUrlPrefixes, updateTypes, matchCurrentVersion). If you wish for configuration to apply to all packages, it is not necessary to place it inside a packageRule at all., npm.packageRules: Each packageRule must contain at least one selector (paths, languages, baseBranchList, managers, datasources, depTypeList, packageNames, packagePatterns, excludePackageNames, excludePackagePatterns, sourceUrlPrefixes, updateTypes, matchCurrentVersion). If you wish for configuration to apply to all packages, it is not necessary to place it inside a packageRule at all.`

I tried `matchUpdateTypes` with/without `matchPackagePatterns` that does not solve the error. 
Can someone help me understand how I need to update my configuration to "valid"?

**Comment by viceice** [comment]
Which version of renovate are you using? Looks like you are using an very outdated one.

**Reply by yevhen-hryhorevskyi** [reply]
We use `whitesource/renovate` `v1.2.0` (docker image) which is published ~5 months ago. Not sure what version that has, but I think it should not be too old.

**Reply by viceice** [reply]
Please check debug log, it should print the renovate version.

@rarkins when do you expect to release a new on premise version?

**Reply by rarkins** [reply]
whitesource/renovate is now up to 1.4.x so please upgrade to that and then update this question accordingly with new logs if necessary

**Reply by yevhen-hryhorevskyi** [reply]
Hi @viceice @rarkins  Thank you for taking a look into this issue. Yes, upgrade to 1.4.x solved the issue. Thank you
